### PR TITLE
[EWL-5018] SG2 | Code standards to-dos for Resource page

### PR DESCRIPTION
**Pull request by JaeBea**
## Ticket(s)

**Jira Ticket**
- [EWL-5018: SG2 | Code standards to-dos for Resource page](https://issues.ama-assn.org/browse/EWL-5018)

## Description
As a developer using the Resource page pattern in SG2, the pattern should be built consistently with Twig and utilize the jQuery UI Tabs library.

This update refactors relevant twig templates, json, js, and scss for resource page tab functionality.

## To Test

- Pull `feature/EWL-5018-code-standards-resource-page` branch
- Run `gulp serve` in ama-style-guide-2/styleguide to ensure a local version of style guide is running.
- Navigate to [Page > Resource](http://localhost:3000/?p=pages-resource).
- Confirm page still works as it did in [Style Guide 2](https://americanmedicalassociation.github.io/ama-style-guide-2/?p=pages-resource)

## Visual Regressions

Updated reference images for resource page backstop tests have been included in this PR.

## Relevant Screenshots/GIFs
N/A

## Remaining Tasks
N/A

## Additional Notes
N/A

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

### CODE REVIEW

**Comment by designerneil** [comment]
The related links tab missing the subtitle under the CME link
screen shot 2018-07-10 at 11 05 20 am
screen shot 2018-07-10 at 11 05 13 am

File: styleguide/source/_patterns/05-pages/resource.twig [hunk 1]
```diff
@@ -4,7 +4,7 @@
 {% include '@organisms/masthead/masthead.twig' %}
 <div class="ama____page--resources__page-content">
   <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad asperiores atque cupiditate <a href="http://www.google.com">normal link</a> debitis dignissimos, dolorem dolorum ea
-  <a href="#" class="ama__page--resource__resource-link" data-resource="downloads-1">(Download 1)</a> eius hic laboriosam, necessitatibus nesciunt nobis omnis placeat reprehenderit tenetur ullam vel veniam. Delectus, eaque est molestiae pariatur quae quidem suscipit tenetur voluptatibus.</p>
```
Comment (designerneil) [inline styleguide/source/_patterns/05-pages/resource.twig]: Can we use the link atom here?

File: styleguide/source/assets/js/tabs.js [hunk 2]
```diff
@@ -1,57 +1,27 @@
...
-    $( ".ama__tabs" ).tabs({
-        activate: function (event, ui) {
-          window.location.hash = ui.newPanel.attr('id');
-          return false;
+    var defaultActiveTab = 0
+    var viewportWidth = window.innerWidth;
+    if (viewportWidth >= 600 && $('.ama__resource-tabs').length > 0) {
+        defaultActiveTab = 1;
+    }
+    $(".ama__tabs").tabs({
```
Comment (designerneil) [inline styleguide/source/assets/js/tabs.js:10]: Well done!

File: styleguide/source/assets/js/tabs.js [hunk 3]
```diff
@@ -1,57 +1,27 @@
...
+            var widget = $(this).data('ui-tabs');
+            $(window).on('hashchange', function() {
+                widget.option('active', widget._getIndex(location.hash));
+            });
         }
-      });
+    });
+
+    // Prevent jump onclick
+    $('.ama__resource-tabs__nav a, .ama__tabs-navigation a').on('click', function(e){
```
Comment (designerneil) [inline styleguide/source/assets/js/tabs.js:21]: Page seems to still jump. I tested it by making my page a bit shorter and clicking on the tabs

**Comment by designerneil** [comment]
screen shot 2018-07-10 at 11 33 44 am

**Comment by designerneil** [comment]
The resource.twig has hard coded text that probably should go into the json file:

```
<div class="ama____page--resources__page-content">
  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad asperiores atque cupiditate <a href="http://www.google.com">normal link</a> debitis dignissimos, dolorem dolorum ea
  <a href="#downloads" class="ama__page--resource__resource-link" >(Download 1)</a> eius hic laboriosam, necessitatibus nesciunt nobis omnis placeat reprehenderit tenetur ullam vel veniam. Delectus, eaque est molestiae pariatur quae quidem suscipit tenetur voluptatibus.</p>
  {% include '@molecules/resource-link-as-inline.twig' with { resourceLinkAsInline : body.resourceLinkAsInline } %}
  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aperiam assumenda consequuntur cupiditate, ea eius, facilis fuga laudantium modi molestiae placeat similique soluta tempora vel? Aliquid cupiditate delectus distinctio, doloribus enim eveniet illum, laudantium obcaecati pariatur perspiciatis quaerat quod similique velit. Eos libero provident quod sit.</p>
</div>
```

**Comment by designerneil** [comment]
Resouce.md need the json structure in it:

```
### Description
A page using the Split layout that houses resources based on the page content.

[EWL-4921](https://issues.ama-assn.org/browse/EWL-4921)

```

**Comment by designerneil** [comment]
The columns seem thinner than in gh-pages causing the header to wrap differently (probably being cause by either a margin or padding around the container, ):

**localhost:**
screen shot 2018-07-10 at 11 40 35 am


**gh-pages:**
screen shot 2018-07-10 at 11 41 33 am

**Comment by JaeBea** [comment]
@designerneil I have addressed all concerns except the jump when clicking a tab. I do not know how to prevent this without removing the hash completely, which causes issues when javascript is not enables. What should work, e.preventDefault(), is not consistently working. I feel we should table that issue and address at a later time as other committed areas using jQueryUI tabs have the same issue.

**Review by designerneil — APPROVED** [review]

**Pull request merged**